contracts/examples/mintable-erc1155-transfer/FxMintableERC1155ChildTunnel.sol

```diff
@@ -0,0 +1,293 @@
+
+        // check if mapping is already there
+        require(rootToChildToken[rootToken] == address(0x0), "FxMintableERC1155ChildTunnel: ALREADY_MAPPED");
+        rootToChildToken[rootToken] = childToken;
+        emit TokenMapped(rootToken, childToken);
+
+        // initialize child token with all parameters
+        FxERC1155(childToken).initialize(address(this), rootToken, _uri);
+    }
+
+    // Mint tokens on child chain
+    function mintToken(
+        address _childToken,
+        uint256 _tokenId,
+        uint256 _amount,
+        bytes calldata _data
+    ) external onlyOwner {
+        FxERC1155 childTokenContract = FxERC1155(_childToken);
+        // child token contract will have root token
+        address rootToken = childTokenContract.connectedToken();
+
+        // validate root and child token mapping
+        require(
+            _childToken != address(0x0) && rootToken != address(0x0) && _childToken == rootToChildToken[rootToken],
+            "FxMintableERC721ChildTunnel: NO_MAPPED_TOKEN"
+        );
+        childTokenContract.mint(msg.sender, _tokenId, _amount, _data);
+    }
+
+    function withdraw(
+        address childToken,
+        uint256 id,
+        uint256 amount,
+        bytes calldata data
+    ) public {
+        _withdraw(childToken, msg.sender, id, amount, data);
+    }
+
+    function withdrawTo(
+        address childToken,
+        address receiver,
+        uint256 id,
+        uint256 amount,
+        bytes calldata data
+    ) public {
+        _withdraw(childToken, receiver, id, amount, data);
+    }
+
+    function withdrawBatch(
+        address childToken,
+        uint256[] calldata ids,
+        uint256[] calldata amounts,
+        bytes calldata data
+    ) public {
+        _withdrawBatch(childToken, msg.sender, ids, amounts, data);
+    }
+
+    function withdrawToBatch(
+        address childToken,
+        address receiver,
+        uint256[] calldata ids,
+        uint256[] calldata amounts,
+        bytes calldata data
+    ) public {
+        _withdrawBatch(childToken, receiver, ids, amounts, data);
+    }
+
+    function onERC1155Received(
+        address, /* operator */
+        address, /* from */
+        uint256, /* tokenId */
+        uint256, /* amount */
+        bytes memory /* data */
+    ) public pure override returns (bytes4) {
+        return this.onERC1155Received.selector;
+    }
+
+    function onERC1155BatchReceived(
+        address, /* operator */
+        address, /* from */
+        uint256[] memory, /* tokenIds */
+        uint256[] memory, /* amounts */
+        bytes memory /* data */
+    ) public pure override returns (bytes4) {
+        return this.onERC1155BatchReceived.selector;
+    }
+
+    //
+    // Internal methods
+    //
+
+    function _processMessageFromRoot(
+        uint256, /* stateId */
+        address sender,
+        bytes memory data
```

Comment:
It can't be used here since this method is called internally by another function decoding `data` in memory and hence passing the argument in memory. Added everywhere else in the external methods: https://github.com/fx-portal/contracts/pull/41/commits/dfcae12e33869b57e67d0bcab42c56c45adaf342 Do let me know if I'm missing something here!